src/MergeContext.js

```diff
@@ -201,10 +201,40 @@ class StatusChecks
         this.addRequiredStatus(new StatusCheck(raw));
     }
 
+    hasAutomatedMergeStatus(state, description) {
+        return this._requiredStatuses.some(el =>
+                el.context.trim() === Config.automatedMergeStatusContext() &&
+                el.state === state &&
+                el.description === description);
```

Comment:
If possible, reorder to place faster and more selective conditions first:

```suggestion
                el.description === description &&
                el.state === state &&
                el.context.trim() === Config.automatedMergeStatusContext();
```